assets/js/modules/analytics-4/datastore/audiences.js

```diff
@@ -203,93 +205,114 @@ const baseActions = {
-				)
-			);
+				const { audienceResourceNames, error } =
+					yield commonActions.await(
+						getNonZeroDataAudiencesSortedByTotalUsers(
+							registry,
+							userAudiences,
+							startDate,
+							endDate
+						)
+					);
+
+				if ( error ) {
+					return { error };
+				}
 
-			if ( ! error ) {
-				// TODO: Full error handling will be implemented via https://github.com/google/site-kit-wp/issues/8134.
 				configuredAudiences.push(
 					...audienceResourceNames.slice( 0, MAX_INITIAL_AUDIENCES )
 				);
 			}
-		}
 
-		if ( configuredAudiences.length < MAX_INITIAL_AUDIENCES ) {
-			// If there are less than two (MAX_INITIAL_AUDIENCES) configured user audiences, add the Site Kit-created audiences if they exist,
-			// up to the limit of two.
+			if ( configuredAudiences.length < MAX_INITIAL_AUDIENCES ) {
+				// If there are less than two (MAX_INITIAL_AUDIENCES) configured user audiences, add the Site Kit-created audiences if they exist,
+				// up to the limit of two.
 
-			const siteKitAudiences = availableAudiences.filter(
-				( { audienceType } ) => audienceType === 'SITE_KIT_AUDIENCE'
-			);
+				const siteKitAudiences = availableAudiences.filter(
+					( { audienceType } ) => audienceType === 'SITE_KIT_AUDIENCE'
+				);
 
-			// Audience slugs to sort by:
-			const sortedSlugs = [ 'new-visitors', 'returning-visitors' ];
+				// Audience slugs to sort by:
+				const sortedSlugs = [ 'new-visitors', 'returning-visitors' ];
 
-			const sortedSiteKitAudiences = siteKitAudiences.sort(
-				( audienceA, audienceB ) => {
-					const indexA = sortedSlugs.indexOf(
-						audienceA.audienceSlug
-					);
-					const indexB = sortedSlugs.indexOf(
-						audienceB.audienceSlug
-					);
+				const sortedSiteKitAudiences = siteKitAudiences.sort(
+					( audienceA, audienceB ) => {
+						const indexA = sortedSlugs.indexOf(
+							audienceA.audienceSlug
+						);
+						const indexB = sortedSlugs.indexOf(
+							audienceB.audienceSlug
+						);
 
-					return indexA - indexB;
-				}
-			);
+						return indexA - indexB;
+					}
+				);
 
-			const audienceResourceNames = sortedSiteKitAudiences
-				.slice( 0, MAX_INITIAL_AUDIENCES - configuredAudiences.length )
-				.map( ( { name } ) => name );
+				const audienceResourceNames = sortedSiteKitAudiences
+					.slice(
+						0,
+						MAX_INITIAL_AUDIENCES - configuredAudiences.length
+					)
+					.map( ( { name } ) => name );
 
-			configuredAudiences.push( ...audienceResourceNames );
+				configuredAudiences.push( ...audienceResourceNames );
+			}
 		}
 
 		if ( configuredAudiences.length === 0 ) {
+			const audiencesToCreate = failedSiteKitAudienceResourceNames?.length
+				? failedSiteKitAudienceResourceNames
+				: [ 'new-visitors', 'returning-visitors' ];
+
 			// If there are no configured audiences by this point, create the "new-visitors" and "returning-visitors" audiences,
 			// and add them to the configured audiences.
 			const [ newVisitorsResult, returningVisitorsResult ] =
```

Comment:
If `audiencesToCreate` only contains a single audience, it could be either "new visitors" or "returning visitors". In this case the current destructuring may not be correct - if we only have "returning visitors", the first element in the returned array will be the result for the "returning visitors" audience. And of course we won't actually have two results.